### ISSUE TITLE
.NET 5 - preparation

### ISSUE THREAD

**Issue by petrsvihlik** [Kentico/kontent-delivery-sdk-net#215]
### Consider System.Text.Json
- improvements: https://devblogs.microsoft.com/dotnet/announcing-net-5-0-preview-3/

**Comment by tomasjurasek** [comment]
I found the extension library for System.Text.Json which supports interface deserialization - https://github.com/dahomey-technologies/Dahomey.Json

**Comment by petrsvihlik** [comment]
Some good news:
- https://devblogs.microsoft.com/dotnet/whats-next-for-system-text-json/

They plan to extend polymorphic serialization and deserialization: https://github.com/dotnet/runtime/issues/45189

So this feature might become unblocked after .NET 6.

**Comment by qpooqp** [comment]
Hello. 
Just want to ask if there will be any progress on this issue now that System.Text.Json got lots of improvements.

I have models generated by the [model generator](https://github.com/kontent-ai/model-generator-net).
My main problem is interface types in the models. For example i have model with property of type `Kontent.Ai.Delivery.Abstractions.ITaxonomyTerm`. 
I need concrete type for deserialization and i don't like the idea that i should create my own types based on the interfaces. That would mean that i need to keep them up to date with the interfaces when they changes and this is additional work. It would be nice to have some concrete types available. 

I was thinking about source generator that would create the concrete types based on the interfaces, but i would prefer to have some "official" solution for me to be able to serialize and deserialize the models.

**Comment by pokornyd** [comment]
This is **fully addressed** in the modernized SDK (v19, `vnext` branch). **Newtonsoft.Json has been completely removed** — the entire SDK uses `System.Text.Json` natively.

### What changed

1. **`System.Text.Json` throughout** — All serialization and deserialization is powered by `System.Text.Json`. No Newtonsoft.Json dependency anywhere in the SDK.

2. **Pluggable deserialization** — The `IContentDeserializer` interface allows you to replace the default `System.Text.Json`-based deserializer with your own implementation if needed (e.g., for a different serializer).

### Concrete types for @qpooqp (thx for the followup!)

The SDK now provides **public concrete types** that the model generator uses directly:

| Interface | Concrete type |
|-----------|--------------|
| `ITaxonomyTerm` | `TaxonomyTerm` (public sealed record) |
| `IAsset` | `Asset` (public sealed record) |
| `IDateTimeContent` | `DateTimeContent` (public sealed record) |
| `IMultipleChoiceOption` | `MultipleChoiceOption` (public sealed record) |

Generated models use these concrete types, so `System.Text.Json` can deserialize them directly — no custom converters, no interface deserialization workarounds:

```csharp
[ContentTypeCodename("article")]
public partial record Article
{
    [JsonPropertyName("personas")]
    public IEnumerable<TaxonomyTerm>? Personas { get; init; }

    [JsonPropertyName("teaser_image")]
    public IEnumerable<Asset>? TeaserImage { get; init; }

    [JsonPropertyName("post_date")]
    public DateTimeContent? PostDate { get; init; }
}
```

No need to create your own concrete types or keep them in sync with interface changes — the SDK provides them out of the box.

---

The SDK is currently pre-released as 19.0.0-rc3, with production release planned for early April, at which point I'll close this issue. It's compatible with the latest beta of [model generator](https://github.com/kontent-ai/model-generator-net). Sync API features are newly released as a [standalone SDK](https://github.com/kontent-ai/sync-sdk-net).
